ecl/eclagent/eclagent.cpp

```diff
@@ -2234,7 +2235,10 @@ void EclAgent::runProcess(IEclProcess *process)
         workflow->perform(this, process);
     }
     else
-        process->perform(this, 0);
+    {
+        GlobalCodeContextExtra gctx(this, 0);
+        process->perform(&gctx, 0);
```

Comment:
I wonder whether if would be better to pass 'this' (as before), and have the EclAgent::getWorkflowId() method return 0;

Reply:
This would improve the code at best minimally.  I'd prefer to leave as is.

Reply:
Am not concerned it is inefficient or problematic as it stands.
It is more a question of under what circumstances can this branch be used, and when it is, does it make sense for ICodeContext::getWorkflowId() to return 0 (and what does that mean?), or should it never be called in which case the EclAgent::getWorkflowId() version that returns throwUnexpected() is more correct.

I am not sure how or when workflow does not exist, so the point is probably moot anyway.

Reply:
I suspect this could be changed so that it throws an error if it does not have workflow.  I think it has always been present for the past 15+ years.